File: ps/best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int i,j=0,res=0;
        for (i=1;i<prices.size();i++){
            if (prices[j] < prices[i]) {
                res+=prices[i]-prices[j];
            }
            j = i;
        }
        return res;
    }
};

File: ps/best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxGap = 0
        stack = []
        for price in prices:
            if not stack:
                stack.append(price)
                continue
            if stack[-1] < price:
                maxGap += price - stack[-1]
            stack.pop()
            stack.append(price)

        return maxGap